packages/miniflare/src/plugins/core/inspector-proxy/inspector-proxy-controller.ts

```diff
@@ -55,11 +60,29 @@ export class InspectorProxyController {
 
 		this.#initializeWebSocketServer(server);
 
+		const listeningPromise = new Promise<void>((resolve) =>
+			server.once("listening", resolve)
+		);
 		server.listen(await this.#inspectorPort);
 
+		await listeningPromise;
+
 		return server;
 	}
 
+	async #restartServer() {
+		const server = await this.#server;
+		server.closeAllConnections();
+		await new Promise<void>((resolve, reject) => {
+			server.close((err) => (err ? reject(err) : resolve()));
+		});
+		const listeningPromise = new Promise<void>((resolve) =>
+			server.once("listening", resolve)
+		);
+		server.listen(await this.#inspectorPort);
```

Comment:
Do we need to re-initialise the websocket server here?

Reply:
oh, that's a great call out! I need to double check that, we probably need to! thanks! 🫶 

(I wanted to keep things simple and didn't add a test that actually uses the websocket, I probably ought to add one!)

Reply:
(mh... but on second thought this worked with fine with my manual testing (where I also tested the devtools page)... mh.... 🤔 maybe this is something that we should do for cleanliness but not completely necessary? 🤔)

Reply:
I re-checked it does seem to work just fine 🙂 

I think `WebSocketServer` is simply coping with the underlying server stopping and re-starting 🙂

Reply:
I've added a test to make sure that this works as intended 🙂 : https://github.com/cloudflare/workers-sdk/pull/8747/commits/a91fd77cc2b6bfbcfc9d5e38e9760d55eb4a7120